### PR TITLE
[CP 2.0] Improve approach to make requests and use Liferay Event Bus API

### PR DESCRIPTION
**Pull request by jairmedeiros**

### CODE REVIEW

**Comment by aildsonf** [comment]
Just starting reviewing :)

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js [hunk 1]
```diff
@@ -62,52 +70,120 @@ const AppContextProvider = ({assetsPath, children, page}) => {
...
+		const fetchData = async () => {
+			const user = await getUser();
 			const projectExternalReferenceCode = SearchParams.get(
 				PARAMS_KEYS.PROJECT_APPLICATION_EXTERNAL_REFERENCE_CODE
 			);
 
-			if (projectExternalReferenceCode) {
-				getProject(projectExternalReferenceCode);
-				getSessionId();
+			if (user && getCurrentPageName() === 'overview') {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:168]: overview should be coming from a constant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js [hunk 2]
```diff
@@ -62,52 +70,120 @@ const AppContextProvider = ({assetsPath, children, page}) => {
...
+				const isValid = await isValidPage(
+					user,
+					projectExternalReferenceCode,
+					'overview'
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:172]: same here

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Pull request closed without merging**